src/main/java/hudson/plugins/jira/pipeline/IssueFieldUpdateStep.java

```diff
@@ -140,19 +137,6 @@ public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskLi
         this.perform(run, workspace, run.getEnvironment(listener), launcher, listener);
     }
 
-    private List<JiraIssueField> expandVariables(List<JiraIssueField> fields, EnvVars envVars) {
```

Comment:
This was interesting, I checked it and fields parameter is always a list with 1 element in it. The logic does not support having multiple fields.

Because, we are asking a field id from user and do some string operation on it such as adding "customField_" prefix etc and creating a JiraIssueField with it and add it to this list.

If user enters multiple field ids, the system wont be able to make any string operation on it and it will fail. So this step always works with 1 field id, so I am not sure why we are creating a list and sending it to expand variables, so I got rid of it as well. Let me know if you think we need it back

Reply:
I checked the test code but we are not sending multiple custom fields to the step. We are sending them one by one and parsing it one by one.

```
    public String prepareFieldId(String fieldId) {
        String prepared = fieldId;
        if (!prepared.startsWith("customfield_"))
            prepared = "customfield_" + prepared;
        return prepared;
    }
```

This is how customfield is parsed in the step, it accepts only String and not String[], so on the test code we are sending the list of String one by one and comparing if it is correct or not each time. In this case it is correct. On the Jenkins interface, we write "customfield_XXX" or "XXX" and step correctly parses it to "customfield_XXX".

So, there is actually no way to send multiple customfields on the step. Step works with only 1 customfield. That is why test passed, because we are not sending multiple fields at the same time.

Reply:
Also another thing I noticed is, during the test we create 2 lists. 1 before the test and 1 after the test. And what we do is sending each element in the "before test" one by one to the step and comparing the outcome with the "after test" list. Here are the lists;

```
    List<String> field_test = Arrays.asList("10100", "customfield_10100", "field_10100");

    List<String> field_after =
        Arrays.asList("customfield_10100", "customfield_10100", "customfield_field_10100");
```

Since test is passing, that means the last item "field_10100" is returning as "customfield_field_10100", but I thought custom fields on Jira are always numerical values. If that is the case, there is no way a field id can be "customfield_field_10100". Altough I am not so sure, so please correct me if I am wrong.